crates/extensions/tedge_downloader_ext/src/actor.rs

```diff
@@ -0,0 +1,108 @@
+use async_trait::async_trait;
+use download::Auth;
+use download::DownloadError;
+use download::DownloadInfo;
+use download::Downloader;
+use log::info;
+use std::path::Path;
+use std::path::PathBuf;
+use tedge_actors::Sequential;
+use tedge_actors::Server;
+use tedge_actors::ServerActorBuilder;
+use tedge_actors::ServerConfig;
+use tedge_utils::file::PermissionEntry;
+
+#[derive(Debug, Clone)]
+pub struct DownloadRequest {
+    pub url: String,
+    pub file_path: PathBuf,
+    pub auth: Option<Auth>,
+}
+
+impl DownloadRequest {
+    pub fn new(url: &str, file_path: &Path, auth: Option<Auth>) -> Self {
+        Self {
+            url: url.into(),
+            file_path: file_path.into(),
+            auth,
+        }
+    }
+}
+
+pub type DownloadResult = Result<DownloadResponse, DownloadError>;
+
+#[derive(Debug)]
+pub struct DownloadResponse {
+    pub url: String,
+    pub file_path: PathBuf,
+}
+
+impl DownloadResponse {
+    fn new(url: &str, file_path: &Path) -> Self {
+        Self {
+            url: url.into(),
+            file_path: file_path.into(),
+        }
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct DownloaderActor {
+    config: ServerConfig,
+}
+
+impl DownloaderActor {
+    pub fn new() -> Self {
+        DownloaderActor::default()
+    }
+
+    pub fn builder(&self) -> ServerActorBuilder<DownloaderActor, Sequential> {
+        ServerActorBuilder::new(DownloaderActor::default(), &ServerConfig::new(), Sequential)
+    }
+
+    pub fn with_capacity(self, capacity: usize) -> Self {
+        Self {
+            config: self.config.with_capacity(capacity),
+        }
+    }
+
+    pub fn with_max_concurrency(self, max_concurrency: usize) -> Self {
```

Comment:
This one is useless I guess?

Reply:
Yes, it's simpler to remove it - as the `config.max_concurrency` will not be used by a `Sequential` server.